### PR TITLE
test(e2e,playwright): side-panel.tsx

### PR DESCRIPTION
**Pull request by 0xLBlaze**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Solves "template-challenge-components/side-panel" #51705 

camperbot cant code

### CODE REVIEW

**Comment by moT01** [comment]
Want to give this an update or rebase @0xLBlaze? One of the tests is failing - I believe [this PR that was merged] (https://github.com/freeCodeCamp/freeCodeCamp/pull/52028) might have fixed the issue.

File: e2e/side-panel.spec.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { test, expect } from '@playwright/test';
+
+test.beforeEach(async ({ page }) => {
+  await page.goto(
+    '/learn/javascript-algorithms-and-data-structures/basic-javascript/declare-javascript-variables'
+  );
+});
+
+test.describe('Challenge Side Panel Component Tests', () => {
```
[the remark this answers is not in the document]
Comment (0xLBlaze) [reply e2e/side-panel.spec.ts]: Removed "Tests" from title.

**Review by huyenltnguyen — COMMENTED** [review]
Since SidePanel has a conditional rendering logic, I think we should have this covered in the test suite as well:

https://github.com/freeCodeCamp/freeCodeCamp/blob/619f16988a8218aac350343b17a01bb6c5e3a862/client/src/templates/Challenges/components/side-panel.tsx#L83

**Comment by rohitrai3** [comment]
> Since SidePanel has a conditional rendering logic, I think we should have this covered in the test suite as well:
> 
> https://github.com/freeCodeCamp/freeCodeCamp/blob/619f16988a8218aac350343b17a01bb6c5e3a862/client/src/templates/Challenges/components/side-panel.tsx#L83

Should I add a test where tool panel is not visible?

**Comment by 0xLBlaze** [comment]
> Since SidePanel has a conditional rendering logic, I think we should have this covered in the test suite as well:
> 
> https://github.com/freeCodeCamp/freeCodeCamp/blob/619f16988a8218aac350343b17a01bb6c5e3a862/client/src/templates/Challenges/components/side-panel.tsx#L83

Should I add a test for tool panel to not be visible?

**Comment by 0xLBlaze** [comment]
Thank you @huyenltnguyen for the info. I have updated the code but now `header.spec.ts` tests are failing which are not related to my tests.

**Review by huyenltnguyen — APPROVED** [review]
LGTM 👍 Thank you @0xLBlaze!

**Review by Ahmadkashif — APPROVED** [review]

**Review by naomi-lgbt — APPROVED** [review]

**Comment by naomi-lgbt** [comment]
Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Pull request merged**